modules/logger/parser/parser.go

```diff
@@ -149,12 +150,11 @@ func parseAndStoreTestOutput(
 				// This must be modified when `logger.DoLog` changes.
 				vals := strings.Split(data, " ")
 				testName := vals[0]
-				logWriter.writeLog(logger, testName, data)
 				previousTestName = testName
+				logWriter.writeLog(logger, testName, data)
 
-			case isIndented && previousTestName != "summary":
-				// In a test result block, so collect the line into all the test results we have seen so far.
-				// Note that previousTestName would only be set to summary if we saw a panic line.
+			case isIndented && isResultLine(data):
```

Comment:
This is a more direct case guard for the specific case that this is trying to capture.